spec/tbd_tests_spec.rb

```diff
@@ -2392,4 +2392,1614 @@
+    expect(io[:edges].size).to eq(66)
+
+    n_edges_at_grade             = 0
+    n_edges_as_balconies         = 0
+    n_edges_as_balconysills      = 0
+    n_edges_as_concave_parapets  = 0
+    n_edges_as_convex_parapets   = 0
+    n_edges_as_rimjoists         = 0
+    n_edges_as_concave_rimjoists = 0
+    n_edges_as_convex_rimjoists  = 0
+    n_edges_as_fenestrations     = 0
+    n_edges_as_heads             = 0
+    n_edges_as_sills             = 0
+    n_edges_as_jambs             = 0
+    n_edges_as_concave_jambs     = 0
+    n_edges_as_convex_jambs      = 0
+    n_edges_as_corners           = 0
+    n_edges_as_concave_corners   = 0
+    n_edges_as_convex_corners    = 0
+    n_edges_as_transitions       = 0
+
+    io[:edges].each do |edge|
+      expect(edge).to have_key(:type)
+
+      n_edges_at_grade             += 1 if edge[:type] == :grade
+      n_edges_at_grade             += 1 if edge[:type] == :gradeconcave
+      n_edges_at_grade             += 1 if edge[:type] == :gradeconvex
+      n_edges_as_balconies         += 1 if edge[:type] == :balcony
+      n_edges_as_balconies         += 1 if edge[:type] == :balconyconcave
+      n_edges_as_balconies         += 1 if edge[:type] == :balconyconvex
+      n_edges_as_balconysills      += 1 if edge[:type] == :balconysill
+      n_edges_as_balconysills      += 1 if edge[:type] == :balconysillconcave
+      n_edges_as_balconysills      += 1 if edge[:type] == :balconysillconvex
+      n_edges_as_concave_parapets  += 1 if edge[:type] == :parapetconcave
+      n_edges_as_convex_parapets   += 1 if edge[:type] == :parapetconvex
+      n_edges_as_rimjoists         += 1 if edge[:type] == :rimjoist
+      n_edges_as_concave_rimjoists += 1 if edge[:type] == :rimjoistconcave
+      n_edges_as_convex_rimjoists  += 1 if edge[:type] == :rimjoistconvex
+      n_edges_as_fenestrations     += 1 if edge[:type] == :fenestration
+      n_edges_as_heads             += 1 if edge[:type] == :head
+      n_edges_as_heads             += 1 if edge[:type] == :headconcave
+      n_edges_as_heads             += 1 if edge[:type] == :headconvex
+      n_edges_as_sills             += 1 if edge[:type] == :sill
+      n_edges_as_sills             += 1 if edge[:type] == :sillconcave
+      n_edges_as_sills             += 1 if edge[:type] == :sillconvex
+      n_edges_as_jambs             += 1 if edge[:type] == :jamb
+      n_edges_as_concave_jambs     += 1 if edge[:type] == :jambconcave
+      n_edges_as_convex_jambs      += 1 if edge[:type] == :jambconvex
+      n_edges_as_corners           += 1 if edge[:type] == :corner
+      n_edges_as_concave_corners   += 1 if edge[:type] == :cornerconcave
+      n_edges_as_convex_corners    += 1 if edge[:type] == :cornerconvex
+      n_edges_as_transitions       += 1 if edge[:type] == :transition
+    end
+
+    expect(n_edges_at_grade            ).to eq( 0)
+    expect(n_edges_as_balconies        ).to eq( 2)
+    expect(n_edges_as_balconysills     ).to eq( 2)
```

Comment:
_Lo Scrigno_ has 2 balconies (one on each side of the longer facades), each broken into 2 separate edges:

- a narrow floor/balcony/fenestration sill
- a longer floor/balcony edge
<img width="1104" alt="balconysills" src="https://github.com/rd2/tbd/assets/98104985/4fd0404a-bbb3-4d9e-b6b3-a3303d1dc54c">

TO DO: make sure subsurface is _fenestrated_.